wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -184,21 +184,21 @@ abstract class Wallet
       scripts <- spksF
       scriptPubKeys =
         utxos.flatMap(_.redeemScriptOpt).toSet ++ scripts.map(_.scriptPubKey)
-      _ <- {
+      blockHashToDownload <- {
         if (scriptPubKeys.isEmpty) {
           //do nothing as an optimization, if we have nothing in the wallet
           //we don't need to search the filters
-          Future.unit
+          Future.successful(Vector.empty)
         } else {
-          FutureUtil.sequentially(blockFilters) {
-            case (blockHash, blockFilter) =>
-              val matcher = SimpleFilterMatcher(blockFilter)
-              if (matcher.matchesAny(scriptPubKeys.toVector.map(_.asmBytes))) {
-                nodeApi.downloadBlocks(Vector(blockHash))
-              } else Future.unit
-          }
+          FutureUtil
+            .batchAndParallelExecute(
```

Comment:
did you test this in visualvm and see an improvement? Just worried about the weird behavior we saw previously and wondering if you might want to use `makeAsync`

Reply:
Yeah, i'm doing so now. Waiting for a block to come in on mainnet so i can do an IBD with this code :+1:

Reply:
> did you test this in visualvm and see an improvement? Just worried about the weird behavior we saw previously and wondering if you might want to use `makeAsync`

You were absolutely right here. Here is what it looks like with `FutureUtil.makeAsync()` :tada: 

![Screenshot from 2021-05-25 11-19-57](https://user-images.githubusercontent.com/3514957/119533707-c168cd80-bd4b-11eb-80cb-aec607f58f8c.png)